system-addon/test/unit/lib/FaviconFeed.test.js

```diff
@@ -163,6 +163,16 @@ describe("FaviconFeed", () => {
       assert.deepEqual(feed._sitesByDomain, data);
       assert.notCalled(feed.cache.set);
     });
+    it("should handle server errors", async () => {
+      feed.loadFromURL = sinon.spy(url => ({data: [], status: 500}));
+      await feed.refresh();
+      assert.equal(1, feed.numRetries);
```

Comment:
is there a good way to test that the delay is set properly? Monkeypatch (erm, mock) `Date.now()`?

Reply:
The `clock = sinon.useFakeTimers()` will have `Date.now()` be `0` and you can advance it however you want. You probably don't need to advance and just check that the "next" time is the expected value, e.g., 0 + 5 minutes = 5 minutes.

Reply:
ah, cool. If I had known, I would've caught my math fail 😉